### PR TITLE
Planner dependency conflicts

### PR DESCRIPTION
**Pull request by juhoeerola**
Implemented in this PR:
- show warning about unmet dependency order
  - tooltip with placeholder link
- show relations of the moved task(s) when dragging task or milestone

Not implemented:
- relation modal linked from the warning tooltip
- correct task measuring (maybe #397)

### CODE REVIEW

**Review by IidaKainu — CHANGES_REQUESTED** [review]
Looks great so far 🔥 
Few minor things could be changed
Furthermore, I found checkBadRelations quite dense (in 'there is a lot happening' way) to read and keep track of what is going on, especially in the end when checking for badRelations. Some of this could be attributed to my lack of sleep though. Still, it would be great if its readability could be improved - maybe by splitting smaller parts to some descriptive functions.

**Comment by juhoeerola** [comment]
Removed unused `possibleVersions` field

File: frontend/src/components/SortableTask.tsx [hunk 1]
```diff
@@ -2,23 +2,77 @@ import { CSSProperties, FC, forwardRef } from 'react';
...
+      <a className="green" href="#todo">
+        Show relations
+      </a>
+    </div>
+  </div>
+);
+
+const RelationIndicator: FC<{ task: TaskProps['task'] }> = ({
+  task: { id, relation, badRelations },
+}) => {
```
[the remark this answers is not in the document]
Comment (juhoeerola) [reply frontend/src/components/SortableTask.tsx:42]: I did consider, but I'm not sure about the effect on readability.

Anyway, switched to earlier null-returns. How do you find that now?

[the remark this answers is not in the document]
Comment (IidaKainu) [reply frontend/src/components/SortableTask.tsx:42]: There's less nesting now, which I think is better 👍

**Review by IidaKainu — APPROVED** [review]
> Removed unused possibleVersions and versionIndex fields, and related code from the bad relation check

> I shuffled things around a little, and named more things

Both of these help with the clarity of checkBadRelations, looks good

**Pull request merged**